nemo/collections/nlp/data/datasets/sgd_dataset/sgd_dataset.py

```diff
@@ -0,0 +1,63 @@
+# =============================================================================
+# Copyright 2020 NVIDIA. All Rights Reserved.
+# Copyright 2019 The Google Research Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+import numpy as np
+from torch.utils.data import Dataset
+
+__all__ = ['SGDDataset']
+
+
+class SGDDataset(Dataset):
```

Comment:
Hmmm, let me try to understand that:
- DataLayer is a wrapper around SGDDataset
- SGDDataset seems to be a wrapper around Dstc8DataProcessor

Moreover, you create each object outside of the other.

Please merge Dstc8DataProcessor with SGDDataset into a single Dataset class.

Reply:
discussed offline